src/engraving/dom/hairpin.cpp

```diff
@@ -268,6 +277,71 @@ Sid HairpinSegment::getPropertyStyle(Pid pid) const
     return TextLineBaseSegment::getPropertyStyle(pid);
 }
 
+Dynamic* HairpinSegment::findDynamic(bool start) const
+{
+    Fraction refTick = start ? hairpin()->tick() : hairpin()->tick2();
+    Measure* measure = score()->tick2measure(start ? refTick : refTick - Fraction::eps());
+    if (!measure) {
+        return nullptr;
+    }
+
+    std::vector<Dynamic*> dynamics;
+    dynamics.reserve(2);
+
+    if (start) {
```

Comment:
It would be better to have two separate methods... findDynamic(true) & findDynamic(false) look confusing...